src/sdk/object_sdk.js

```diff
@@ -443,9 +452,16 @@ class ObjectSDK {
                 if (ranges.length !== 1) throw new Error('fix_copy_source_params: multiple ranges not supported');
                 source_params.start = ranges[0].start;
                 source_params.end = ranges[0].end;
+                params.copy_source.bucket = actual_source_ns.get_bucket(bucket);
+                params.copy_source.obj_id = source_md.obj_id;
+                params.size = source_params.end - source_params.start;
             }
+
+            source_params.object_md = source_md;
+            source_params.obj_id = source_md.obj_id;
+
             params.source_stream = await source_ns.read_object_stream(source_params, this);
-            if (params.size > (100 * size_utils.MEGABYTE)) {
+            if (params.size && params.size > (100 * size_utils.MEGABYTE)) {
```

Comment:
Can you explain what did you try to achieve with these additions? 
It doesn't look right to me - you use copy_source when not the same namespace?

Reply:
```
                params.copy_source.bucket = actual_source_ns.get_bucket(bucket);
                params.copy_source.obj_id = source_md.obj_id;
                params.size = source_params.end - source_params.start;
```
This was added for a case where a byte range is copied from an object bucket into a namespace bucket.
The test is at: https://github.com/noobaa/noobaa-core/blob/684f4e672eda381e9f63159c51d3c5a9cf61af79/src/test/unit_tests/test_s3_ops.js#L384

Reply:
@guymguym 
```
            source_params.object_md = source_md;
            source_params.obj_id = source_md.obj_id;
```
Added because of exceptions thrown when 
https://github.com/noobaa/noobaa-core/blob/684f4e672eda381e9f63159c51d3c5a9cf61af79/src/test/unit_tests/test_s3_ops.js#L344
is executed.

Reply:
The tests which give this exception are
https://github.com/noobaa/noobaa-core/blob/684f4e672eda381e9f63159c51d3c5a9cf61af79/src/test/unit_tests/test_s3_ops.js#L509
https://github.com/noobaa/noobaa-core/blob/684f4e672eda381e9f63159c51d3c5a9cf61af79/src/test/unit_tests/test_s3_ops.js#L513

Reply:
so we want to remove these changes and see why the test is failing,
because it might be a new test case that is the cause